### DISCUSSION TITLE
All websockets close and do not reconnect

### DISCUSSION THREAD

**Discussion by austin43** [expo/expo#17089] [Archived: Questions]
I am unsure if others can reproduce this so figured I'd start a discussion.

When starting a brand new app using expo-cli@5.3.1/SDK 44 I am able to connect initially and begin development.  Once I minimize the app, lock the phone, then come back I receive

```Cannot connect to Metro.

Try the following to fix the issue:
- Ensure that Metro is running and available on the same network
- Ensure that the Metro URL is correctly set in AppDelegate

URL: 192.168.86.26:8081

Error: The operation couldn’t be completed. Socket is not connected
at node_modules/metro-runtime/src/modules/vendor/eventemitter3.js:195:15 in emit
at node_modules/metro-runtime/src/modules/HMRClient.js:141:6 in _ws.onerror
at node_modules/react-native/Libraries/WebSocket/WebSocket.js:258:8 in _eventEmitter.addListener$argument_1
at node_modules/react-native/Libraries/vendor/emitter/_EventEmitter.js:135:10 in EventEmitter#emit
```

This in itself isn't a big deal as I can just reload the dev app and continue, but I realized something deeper is potentially going on with websockets.  Our app uses [graphql-ws](https://github.com/enisdenjo/graphql-ws) to open websockets for graphql subscription handlers using [urql](https://formidable.com/open-source/urql/).

Upon minimizing the app, locking my phone, then opening it up after a few seconds, the subscriptions stop working because the websocket closes and will not re-open. 

My hypothesis is this is a deeper issue with websockets not working properly in expo/react-native.

Steps:

```
expo init
yarn start
minimize app (not close)
lock phone
unlock phone
open app
see metro error
```

**Comment by brentvatne** [comment]
is this in expo go?

**Comment by austin43** [comment]
@brentvatne This is in an EAS custom dev build.  I was able to get graphql-ws to reconnect by using their new shouldRetry(), but this feels like a timing issue with websockets when the app awakens.  Sorry for the lack of repro case here, I can try to get one spun up if it helps.

https://github.com/enisdenjo/graphql-ws/commit/9d5c5733ba8bb181ef4f2051da958d80786c9ee6
